### PR TITLE
Add bootstrap command

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#13206

This adds the bootstrap command. Currently this

- Creates the application group CRDs
- Creates the giantswarm-critical priority class. In prod this is created via ignition.
- Creates the giantswarm namespace.

Next PR will add installing app-operator and chart-operator using our helmclient.

```
$ kind create cluster
$ apptestctl bootstrap --kubeconfig="$(kind get kubeconfig)" | luigi

Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.18.2) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Thanks for using kind! 😊
D 09/14 13:40:56 creating REST config from kubeconfig | k8sclient/v4/pkg/k8srestconfig/rest_config.go:137
D 09/14 13:40:56 created REST config from kubeconfig | k8sclient/v4/pkg/k8srestconfig/rest_config.go:145
D 09/14 13:40:56 ensuring `AppCatalog` CRD exists | apptestctl/cmd/bootstrap/runner.go:102
D 09/14 13:40:56 creating CRD `appcatalogs.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:89
D 09/14 13:40:56 created CRD `appcatalogs.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:100
D 09/14 13:40:56 updating CRD `appcatalogs.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:124
D 09/14 13:40:57 updating CRD `appcatalogs.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:124
D 09/14 13:40:57 updated CRD `appcatalogs.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:137
D 09/14 13:40:58 ensured `AppCatalog` CRD exists | apptestctl/cmd/bootstrap/runner.go:109
D 09/14 13:40:58 ensuring `App` CRD exists | apptestctl/cmd/bootstrap/runner.go:102
D 09/14 13:40:58 creating CRD `apps.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:89
D 09/14 13:40:58 created CRD `apps.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:100
D 09/14 13:40:58 updating CRD `apps.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:124
D 09/14 13:40:59 updating CRD `apps.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:124
D 09/14 13:40:59 updated CRD `apps.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:137
D 09/14 13:40:59 ensured `App` CRD exists | apptestctl/cmd/bootstrap/runner.go:109
D 09/14 13:40:59 ensuring `Chart` CRD exists | apptestctl/cmd/bootstrap/runner.go:102
D 09/14 13:40:59 creating CRD `charts.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:89
D 09/14 13:40:59 created CRD `charts.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:100
D 09/14 13:40:59 updating CRD `charts.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:124
D 09/14 13:41:00 updating CRD `charts.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:124
D 09/14 13:41:00 updated CRD `charts.application.giantswarm.io` | k8sclient/v4/pkg/k8scrdclient/crd_client.go:137
D 09/14 13:41:00 ensured `Chart` CRD exists | apptestctl/cmd/bootstrap/runner.go:109
D 09/14 13:41:00 creating priorityclass `giantswarm-critical` | apptestctl/cmd/bootstrap/runner.go:164
D 09/14 13:41:00 created priorityclass `giantswarm-critical` | apptestctl/cmd/bootstrap/runner.go:182
D 09/14 13:41:00 ensuring namespace `giantswarm` | apptestctl/cmd/bootstrap/runner.go:119
D 09/14 13:41:00 ensured namespace `giantswarm` | apptestctl/cmd/bootstrap/runner.go:156
```

### CODE REVIEW

File: cmd/bootstrap/runner.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	k8sClientConfig := k8sclient.ClientsConfig{
+		Logger:     r.logger,
+		RestConfig: restConfig,
+	}
+	k8sClients, err := k8sclient.NewClients(k8sClientConfig)
```
Comment (tomahawk28) [inline cmd/bootstrap/runner.go:68]:
```suggestion
	}
	
	k8sClients, err := k8sclient.NewClients(k8sClientConfig)
```
Align with others, newline in here.

File: cmd/bootstrap/runner.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	// Ensure Application group CRDs are created.
+	crds := []string{
+		"AppCatalog",
+		"App",
+		"Chart",
+	}
+
+	{
+		for _, crdName := range crds {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring %#q CRD exists", crdName))
```
Comment (tomahawk28) [inline cmd/bootstrap/runner.go:101]:
```suggestion
			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring %#q CRD", crdName))
```
Why not without the `exists` verb?

File: cmd/bootstrap/runner.go (hunk 1 above)
Comment (rossf7) [reply cmd/bootstrap/runner.go]: This was copy pasta I updated it to use our usual block style. I think it looks better now but PTAL.

File: cmd/bootstrap/runner.go (hunk 2 above)
Comment (rossf7) [reply cmd/bootstrap/runner.go]: This was also copy pasta I agree reads better without it. Done.

**Review by tomahawk28 — APPROVED** [review]
LGTM!

**Pull request merged**